### PR TITLE
Code splitting POC

### PR DESCRIPTION
**Pull request by leehart**
This seems to be one way forward. (Not sure if it's right though.)

### CODE REVIEW

File: webapp/src/js/components/Plot.js [hunk 1]
```diff
@@ -32,7 +30,24 @@ let Plot = createReactClass({
     };
   },
 
+  componentWillMount() {
+    Promise.all([
```
Comment (benjeffery) [inline webapp/src/js/components/Plot.js]: You don't need `.all` here as you only have a single promise.

File: webapp/src/js/components/Plot.js [hunk 2]
```diff
@@ -32,7 +30,24 @@ let Plot = createReactClass({
...
+      import('plotly.js/dist/plotly-cartesian'),
+      //...more imports
+    ]).then(([Plotly]) => {
+      this.hasImported = true;
+      this.Plotly = Plotly;
+      this.PlotlyComponent = createPlotlyComponent(this.Plotly);
+      this.forceUpdate();
```
Comment (benjeffery) [inline webapp/src/js/components/Plot.js]: Why not store plotly component in state?
`this.setState({plotlyComponent: createPlotlyComponent(Plotly)})
and then use in render?

File: webapp/src/js/components/Plot.js [hunk 3]
```diff
@@ -32,7 +30,24 @@ let Plot = createReactClass({
...
+    }).catch((error) => 'An error occurred while loading the component: ' + error);
+  },
+
   render() {
+
+    if (!this.hasImported) {
```
Comment (benjeffery) [inline webapp/src/js/components/Plot.js]: We could render `<Loading>` here instead of being blank.

File: webapp/src/js/components/Plot.js (hunk 1 above)
Comment (leehart) [reply webapp/src/js/components/Plot.js]: Yup, it's just to demonstrate a mechanism for multiple dynamic imports. Hence the comment on the next line.

File: webapp/src/js/components/Plot.js (hunk 2 above)
Comment (leehart) [reply webapp/src/js/components/Plot.js]: Good question; I don't know. What are the pros and cons?

File: webapp/src/js/components/Plot.js (hunk 3 above)
Comment (leehart) [reply webapp/src/js/components/Plot.js]: Cool, nice idea.

File: webapp/src/js/components/Plot.js (hunk 2 above)
Comment (leehart) [reply webapp/src/js/components/Plot.js]: Yes, I think that will generally lead to cleaner code; trying it out now.

File: webapp/src/js/components/Plot.js (hunk 1 above)
Comment (leehart) [reply webapp/src/js/components/Plot.js]: I'll dynamically import `react-plotlyjs` as well, just for kicks.

Comment (benjeffery) [reply webapp/src/js/components/Plot.js]: Ahh ok, makes sense

**Pull request merged**